plugins/dev-tools/src/playground/index.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Entry point for the Blockly playground.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+
+import * as Blockly from 'blockly/core';
+import * as BlocklyJS from 'blockly/javascript';
+import * as BlocklyPython from 'blockly/python';
+import * as BlocklyLua from 'blockly/lua';
+import * as BlocklyDart from 'blockly/dart';
+import * as BlocklyPHP from 'blockly/php';
+
+import {renderPlayground, renderCheckbox, renderCodeTab} from './ui';
+import {addCodeEditor} from './monaco';
+import addGUIControls from '../addGUIControls';
+
+/**
+ * @typedef {function(!HTMLElement,!Blockly.BlocklyOptions):Blockly.Workspace}
+ */
+let CreateWorkspaceFn;
+
+/**
+ * @typedef {{
+ *     addAction: function(string,function(!Blockly.Workspace):void,string=),
+ *     addGenerator: function(string,!Blockly.Generator,string=),
+ * }}
+ */
+let PlaygroundAPI;
+
+/**
+ * Create the Blockly playground.
+ * @param {!HTMLElement} container Container element.
+ * @param {CreateWorkspaceFn} createWorkspace A workspace creation method called
+ *     every time the toolbox is re-configured.
+ * @param {Blockly.BlocklyOptions} defaultOptions The default workspace options
+ *     to use.
+ * @param {string=} vsEditorPath Optional editor path.
+ * @return {Promise<PlaygroundAPI>} A promise to the playground API.
+ */
+export function createPlayground(container, createWorkspace,
+    defaultOptions, vsEditorPath) {
+  const {blocklyDiv, monacoDiv, guiContainer, tabButtons, tabsDiv} =
+    renderPlayground(container);
+
+  // Load the code editor.
+  return addCodeEditor(monacoDiv, {
+    model: null,
+    language: 'xml',
+    minimap: {
+      enabled: false,
+    },
+    theme: 'vs-dark',
+    scrollBeyondLastLine: false,
+    automaticLayout: true,
+  }, vsEditorPath).then((editor) => {
```

Comment:
Use helper functions--this single function is hard to read.

Reply:
Done.